src/linter/manifestJson/ManifestLinter.ts

```diff
@@ -89,16 +120,59 @@ export default class ManifestLinter {
 		}
 
 		// Detect deprecated view type in routing.config:
-		if (routing?.config && routing.config.viewType && deprecatedViewTypes.includes(routing.config.viewType)) {
+		if (routing?.config?.viewType && deprecatedViewTypes.includes(routing.config.viewType)) {
 			this.#reporter?.addMessage(MESSAGE.DEPRECATED_VIEW_TYPE, {
 				viewType: routing.config.viewType,
 			}, "/sap.ui5/routing/config/viewType");
 		}
 
 		// Detect deprecations in routing.targets:
 		const targets = routing?.targets;
+		const oldToNewTargetPropsMap = {
+			viewName: "name",
+			viewId: "id",
+			viewLevel: "level",
+			viewPath: "path",
+		};
+
+		if (isManifest2 && routing?.config) {
+			for (const [oldProp, newProp] of Object.entries(oldToNewTargetPropsMap)) {
+				if (routing.config[oldProp] && !routing.config[newProp]) {
+					this.#reporter?.addMessage(MESSAGE.NO_RENAMED_MANIFEST_PROPERTY, {
+						propName: oldProp,
+						newName: newProp,
+					}, `/sap.ui5/routing/config/${oldProp}`);
+				}
+			}
+		}
+
 		if (targets) {
+			const configType = routing?.config?.type as string | undefined;
+
 			for (const [key, target] of Object.entries(targets)) {
+				if (isManifest2) {
+					for (const [oldProp, newProp] of Object.entries(oldToNewTargetPropsMap)) {
+						if (target[oldProp] && !target[newProp]) {
+							this.#reporter?.addMessage(MESSAGE.NO_RENAMED_MANIFEST_PROPERTY, {
+								propName: oldProp,
+								newName: newProp,
+							}, `/sap.ui5/routing/targets/${key}/${oldProp}`);
+						}
+					}
+
+					if (configType === undefined && target?.type === undefined) {
+						// Type must be defined somewhere, either in routing.config or in every target.type
```

Comment:
This comment explains what to do, but I think that is missing in the message. When there are several targets, the current behavior is that for EVERY target you get a message asking to add the property to the target. But usually, IMHO it's easier and better to add the `type: "View"` to the general config, and use `type: "Component"` on targets where a component instead of a view is targeted.

cc @stopcoder